misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/CuratorFrameworkProvider.kt

```diff
@@ -12,6 +12,15 @@ internal class CuratorFrameworkProvider @Inject internal constructor(
 ) : Provider<CuratorFramework> {
 
   override fun get(): CuratorFramework {
+    if (config.cert_store != null && config.trust_store != null) {
+      System.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty")
```

Comment:
you _have_ to configure via system properties? you can't configure the zkclient directly?

Reply:
from a quick search, I _think_ you can configure it via the `ZkClientConfig` using the `setProperty`

Reply:
Yeah, I looked into this for a bit previously and the problem is that Curator wraps the ZK client and doesn't give you anyway to set these configuration options.

Reply:
it looks like you can set a `ZookeeperFactory` when building the `CuratorFrameworkFactory`. This should allow configuring the `ZkClientConfig`

Reply:
```
    return CuratorFrameworkFactory.builder()
        .connectString(config.zk_connect)
        .zookeeperFactory { connectString, sessionTimeout, watcher, canBeReadOnly ->
          val zk = DefaultZookeeperFactory().newZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly)
          zk.clientConfig.setProperty("foo", "bar")
          zk
        }
```

Reply:
Hah, wow. Nice find.